internal/provider/service_resource.go

```diff
@@ -353,6 +357,16 @@ func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest
 		}
 		return
 	}
+
+	// Check if a user-specified password was provided and update it if so
+	if !plan.Password.IsNull() && plan.Password.ValueString() != response.InitialPassword {
+		err = r.client.ResetServicePassword(ctx, service.ID, plan.Password.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("Password Update Failed", fmt.Sprintf("Unable to update password, got error: %s", err))
```

Comment:
Nit: from the user perspective, this is not update, as the user provided the password with the new service. Perhaps we write "Setting the password failed". 

Aso, I wonder if we shall not treat this as Fatal, and return the original password the service has been created with.

Reply:
Good point. I added removal of database in case password changing failed. I guess during service creation if we fail setting the password in the database that's indeed what we should be doing because users would expect that database is running with the password that they specified. Also, just failing without cleaning up after ourselves leaves orphaned resources, which shouldn't happen.